debezium-core/src/main/java/io/debezium/transforms/reselect/DefaultReselectColumnsMetadataProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.reselect;
+
+import java.util.List;
+
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.debezium.relational.TableId;
+
+/**
+ * Default re-select columns metadata provider implementation.
+ *
+ * This implementation simply fetches the current re-select column's state from the current row, if exists,
+ * and performs no conversion of the re-selected column.
+ *
+ * @author Chris Cranford
+ */
+public class DefaultReselectColumnsMetadataProvider<R extends ConnectRecord<R>> implements ReselectColumnsMetadataProvider<R> {
+
+    @Override
+    public String getName() {
+        return "default";
+    }
+
+    @Override
+    public String getQuery(R record, Struct source, List<String> columns, TableId tableId) {
+        final String tableName = String.format("%s.%s", tableId.schema(), tableId.table());
+        final StringBuilder query = new StringBuilder();
+        query.append("SELECT ").append(String.join(",", columns));
+        query.append(" FROM ").append(tableName);
+        query.append(" WHERE ").append(createPrimaryKeyWhereClause(record));
+        return query.toString();
+    }
+
+    @Override
+    public Object convertValue(int jdbcType, Object value) {
+        // todo: we need to consider how to inject ValueConverters into this :(
+        return value;
+    }
+
+    protected String createPrimaryKeyWhereClause(R record) {
+        Object keyValue = record.key();
```

Comment:
We need a proper strategy or documentation in a sense of how `message.key.columns` setting can impact this.